### ISSUE TITLE
Form not scrolling to first error when the nested form field is Form.List

### ISSUE THREAD

**Issue by gges5110** [ant-design/ant-design#26197]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/nostalgic-curie-bb8cx?file=/index.js)

### Steps to reproduce
1. Construct a Form with scrollToFirstError set to true
2. Wrap a Form.List with Form.Item
3. Add some rules to the Form.Item e.g. 
```tsx
rules={[
  {
    min: 1,
    message: "at least one element is required",
    type: "array",
    required: true
  }
]}
```
4. When submit, any error happening to the Form.Item will not cause the Form to scroll to the error from the Form.Item that's wrapping around Form.List

### What is expected?
The Form should scroll to the error triggered by Form.Item

### What is actually happening?
When submit, any error happening to the Form.Item will not cause the Form to scroll to the error from the Form.Item that's wrapping around Form.List

| Environment | Info |
|---|---|
| antd | 4.5.4 |
| React | 16.12.0 |
| System | macOS 10.14.6 |
| Browser | Chrome 84.0.4147.105 |

---
Please let me know if I am using Form.List in the wrong way.

**Comment by zombieJ** [comment]
Your reproduce seems auto scroll is workable:
![Kapture 2020-08-14 at 11 58 41](https://user-images.githubusercontent.com/5378891/90212292-900abb00-de25-11ea-8b4b-63fc8e77e8cd.gif)

**Comment by gges5110** [comment]
I am talking about the rule defined by the outer form item (i.e. user list requires at least one element), not the required user first name or last name

**Comment by gges5110** [comment]
I've updated the sandbox, I hope this removes the confusion

**Comment by zombieJ** [comment]
Form.List is a field which not support rules yet. Merge into https://github.com/react-component/field-form/issues/168

**Comment by gges5110** [comment]
The rules are actually working correctly since it's not added to Form.List but the outer Form.Item. You can see the error showing up when you hit the submit button, the only issue is the form is not scrolling to it. That makes me think that this is not an issue caused by Form.List lacking the support of rules, but maybe Form.List couldn't be registered as a form field like \<Input\>.
![image](https://user-images.githubusercontent.com/11253169/90258944-db659f00-de0e-11ea-9e5d-53911e251b70.png)

**Comment by gges5110** [comment]
I've found a workaround for this by adding a div element between Form.List and Form.Item, but it would be nice to not have to do this.

```tsx
<Form
  name="dynamic_form_nest_item"
  onFinish={onFinish}
  autoComplete="off"
  scrollToFirstError={true}
>
  <Form.Item
    name="users"
    rules={[
      {
        min: 1,
        message: "at least one element is required",
        type: "array",
        required: true
      }
    ]}
  >
    {/* placing a div here such that Form can identify this field */}
    <div>
      <Form.List name="users">
        {(fields, { add, remove }) => {
          return (
            ...
          );
        }}
      </Form.List>
    </div>
  </Form.Item>
  <div style={{ height: 1500 }} />

  <Form.Item>
    <Button type="primary" htmlType="submit">
      Submit
    </Button>
  </Form.Item>
</Form>
```

**Comment by zombieJ** [comment]
It's caused that `onChange` propagation from nest children whose behavior is not a promise that will work in any patch update.

**Comment by rfgonzalezweb** [comment]
TL;DR, 
From a custom input inside the Form.Item, take the **id** prop coming from the Form.Item and render it on the JSX

demo in stackblitz
https://react-ydr6fz-aqkkzi.stackblitz.io
